test/sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncConfigTests.kt

```diff
@@ -498,6 +508,81 @@ class SyncConfigTests {
+            FlexParentObject::class,
+            FlexChildObject::class,
+            FlexEmbeddedObject::class,
+        )
+
+        val validateConfig = { config: io.realm.kotlin.Configuration ->
+            assertEquals(3, config.schema.size)
+        }
+
+        // Partition-based Sync
+        enumValues<ValueType>().forEach {
+            when (it) {
+                ValueType.STRING -> {
+                    validateConfig(SyncConfiguration.create(user, "", supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, "", supportedSchemaTypes).build())
+                }
+                ValueType.INT -> {
+                    validateConfig(SyncConfiguration.create(user, 1 as Int, supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, 1 as Int, supportedSchemaTypes).build())
+                }
+                ValueType.LONG -> {
+                    validateConfig(SyncConfiguration.create(user, 1L, supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, 1L, supportedSchemaTypes).build())
+                }
+                ValueType.OBJECT_ID -> {
+                    validateConfig(SyncConfiguration.create(user, ObjectId.create(), supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, ObjectId.create(), supportedSchemaTypes).build())
+                }
+                // ValueType.UUID -> {
+                //     validateConfig(SyncConfiguration.create(user, null as RealmUUID, supportedSchemaTypes))
+                //     validateConfig(SyncConfiguration.Builder(user, null as RealmUUID, supportedSchemaTypes).build())
+                // }
+                ValueType.NULL -> {
+                    validateConfig(SyncConfiguration.create(user, null as String?, supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, null as String?, supportedSchemaTypes).build())
+                }
+                else -> TODO("Test for partition type not defined")
+            }
+
+            // Flexible Sync
+            validateConfig(SyncConfiguration.create(user, supportedSchemaTypes))
+            validateConfig(SyncConfiguration.Builder(user, supportedSchemaTypes).build())
+        }
+
+        // Only do a smoke test for one constructor type as they all go through the same
+        // backing code and testing all combinations didn't seem worth.
+        val config1 = SyncConfiguration.create(user, "", supportedSchemaTypes)
+        assertEquals(3, config1.schema.size)
+        val config2 = SyncConfiguration.Builder(user, "", supportedSchemaTypes).build()
+        assertEquals(3, config2.schema.size)
+    }
+
+    @Test
+    fun unsupportedSchemaTypesThrowException() {
+        val user = createTestUser()
+        val unsupportedSchemaType = setOf(DynamicRealmObject::class)
+
+        // There is no way for us to detect if a KClass implements a certain interface, this is
```

Comment:
We could easily do a public predicate method around https://github.com/realm/realm-kotlin/blob/master/packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmObjectUtil.kt#L105 that would tell if a KClass was of `BaseRealmObject`. If not in this PR we could just make an issue to track it.

Reply:
Ah yeah. I could use `realmObjectCompanionOrNull` as a test for this 👍